### ISSUE TITLE
Sniff to check for the unused parameters in hooks

### ISSUE THREAD

**Issue by dingo-d** [WordPress/WordPress-Coding-Standards#1882]
## Is your feature request related to a problem?

A lot of times plugin and theme authors add hook callbacks but fail to provide all the necessary parameters. For instance, in custom widgets (taken from the [WPThemeReview issue](https://github.com/WPTRT/WPThemeReview/issues/42)):

```php
$title = apply_filters( 'widget_title', $title );
```

and according to the [documentation](https://developer.wordpress.org/reference/hooks/widget_title/), there should exist two extra parameters:

```php
$title = apply_filters( 'widget_title', $title, $instance, $id_base );
```

## Describe the solution you'd like

A solution would be to have a sniff that checks if any of the core WP hooks, that are invoked in the plugin or theme, have the correct amount of parameters passed as in the hook definition.

We would rely on the utility sniff that would check for all the core hooks, which would be a part of #1803 issue
